pos_payment_terminal/models/pos_order.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# © 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from collections import defaultdict
+import logging
+
+from odoo import models, api
+
+_logger = logging.getLogger(__name__)
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    @api.model
+    def _match_transactions_to_payments(self, pos_order):
+        payments = pos_order['statement_ids']
+        transactions = pos_order['transactions']
+        card_journals = self.env['account.journal'].search([
+            ('id', 'in', [p[2]['journal_id'] for p in payments]),
+            ('payment_mode', '!=', False),
+        ])
+        card_payments = [record[2] for record in payments
+                         if record[2]['journal_id'] in card_journals.ids]
+
+        def amount_cents(obj):
+            if 'amount_cents' in obj:
+                return obj['amount_cents']
+            else:
+                return int(round(obj['amount'] * 100))
```

Comment:
here, I guess it does not manage currencies with decimals <>2  (ie XPF)

Reply:
I believe you're right, I'll push an update using the currency's number of decimals. Thanks.